cleverpush/src/main/java/com/cleverpush/database/TableBannerTrackEventDao.java

```diff
@@ -0,0 +1,38 @@
+package com.cleverpush.database;
+
+import androidx.room.Dao;
+import androidx.room.Insert;
+import androidx.room.Query;
+import androidx.room.Update;
+
+import java.util.List;
+
+@Dao
+public interface TableBannerTrackEventDao {
+
+  @Query("SELECT * FROM TableBannerTrackEvent ORDER BY Id ASC")
+  List<TableBannerTrackEvent> getAll();
+
+  @Insert
+  void insert(TableBannerTrackEvent userTbl);
+
+  @Update
+  void update(TableBannerTrackEvent bannerTrackEvent);
+
+  @Query("SELECT * FROM TableBannerTrackEvent WHERE bannerId = :bannerId AND trackEventId = :trackEventId "
+          + "AND property = :property AND value = :value AND relation = :relation AND fromValue = :fromValue AND toValue = :toValue")
+  List<TableBannerTrackEvent> getBannerTrackEvent(String bannerId, String trackEventId, String property, String value, String relation, String fromValue, String toValue);
+
+  @Query("SELECT * FROM TableBannerTrackEvent WHERE trackEventId = :trackEventId ")
+  List<TableBannerTrackEvent> getBannerTrackEvent(String trackEventId);
+
+  @Query("UPDATE TableBannerTrackEvent SET count = count + 1, updatedDateTime = :updatedDate WHERE trackEventId = :eventId")
+  void updateCount(String eventId, String updatedDate);
+
+  @Query("DELETE FROM TableBannerTrackEvent")
+  void deleteAll();
+
+  @Query("DELETE FROM TableBannerTrackEvent WHERE createdDateTime <= strftime('%Y-%m-%d %H:%M:%S', datetime('now', '-' || :retentionDays || ' days'))")
```

Comment:
'-' || :retentionDays || ' days')
this is not logical or operator or?

Reply:
`strftime('%Y-%m-%d %H:%M:%S', datetime('now', '-' || :retentionDays || ' days'))`: This part calculates a date and time in the format 'YYYY-MM-DD HH:MM:SS' that is a certain number of days before the current time.

`datetime('now', '-' || :retentionDays || ' days')`: This portion calculates the current date and time (using 'now') and then subtracts the number of days specified by the retentionDays parameter.

the `||` operator is used to concatenate strings. The `||` operator concatenates the strings to form a single string representing a duration in the format expected by the `datetime` function. For example, if `:retentionDays` is 7, then the result would be a string like '-7 days', and this string is used to calculate a date and time relative to the current moment.

Reply:
normal string concatenation doesnt work here?